src/sentry/integrations/msteams/card_builder/base.py

```diff
@@ -0,0 +1,118 @@
+from abc import ABC
+from enum import Enum
+from typing import Any, Optional, Sequence
+
+from sentry.integrations.notifications import AbstractMessageBuilder
+from sentry.utils.assets import get_asset_url
+from sentry.utils.http import absolute_uri
+
+SENTRY_ICON_URL = "images/sentry-glyph-black.png"
+
+# NOTE: The classes below need to inherit from `str` as well to be serialized correctly.]
+
+
+class TextSize(str, Enum):
+    SMALL = "Small"
+    MEDIUM = "Medium"
+    LARGE = "Large"
+
+
+class TextWeight(str, Enum):
+    BOLDER = "Bolder"
+    LIGHTER = "Lighter"
+
+
+class ActionType(str, Enum):
+    OPEN_URL = "Action.OpenUrl"
+    SUBMIT = "Action.Submit"
+    SHOW_CARD = "Action.ShowCard"
+
+
+class ColumnWidth(str, Enum):
+    STRECH = "strech"
+    AUTO = "auto"
+
+
+class VerticalContentAlignment(str, Enum):
+    CENTER = "Center"
+
+
+REQUIRED_ACTION_PARAM = {
+    ActionType.OPEN_URL: "url",
+    ActionType.SUBMIT: "data",
+    ActionType.SHOW_CARD: "card",
+}
+
+
+class MSTeamsMessageBuilder(AbstractMessageBuilder, ABC):  # type: ignore
+    def build(self) -> Any:
+        """Abstract `build` method that all inheritors must implement."""
+        raise NotImplementedError
+
+    @staticmethod
+    def get_text_block(text: str, **kwargs: str) -> Any:
+        return {"type": "TextBlock", "text": text, "wrap": True, **kwargs}
+
+    def get_logo_block(self) -> Any:
+        return self.get_image_block(get_asset_url("sentry", SENTRY_ICON_URL))
+
+    @staticmethod
+    def get_image_block(url: str) -> Any:
+        return {
+            "type": "Image",
+            "url": absolute_uri(url),
+            "size": "Medium",
```

Comment:
Probably should be `TextSize.Medium` right?

Reply:
TextSize and ImageSize are different. Added ImageSize.